### ISSUE TITLE
Remove `default` from `result.level` property in schema

### ISSUE THREAD

**Issue by lukecartey** [oasis-tcs/sarif-spec#105]
The `result.level` property is defined in the schema with a `default` value of `warning`:
```
        "level": {
          "description": "A value specifying the severity level of the result. If this property is not present, its implied value is 'warning'.",
          "default": "warning",
          "enum": [ "notApplicable", "pass", "note", "warning", "error" ]
        },
```
How this is interpreted depends on the tool that is processing the schema. We happen to use http://www.jsonschema2pojo.org/ to generate Java class files, which produces the following Java field definition:

```
    @JsonProperty("level")
    @JsonPropertyDescription("A value specifying the severity level of the result. If this property is not present, its implied value is 'warning'.")
    private Result.Level level = Result.Level.fromValue("warning");
```
This is a very reasonable interpretation of the schema. However, it has two consequences:

 1. When _loading_ SARIF files where the `result.level` property is missing, the `result.level` property will be automatically set to `warning`.
 1. When _creating_ SARIF result objects, the `result.level` will be explicitly set to `warning` in the output file, unless the `result.level` property is explicitly unset.

This seems to be at odds with the specification itself. The relevant part:

> If the level property is absent, its value SHALL be considered to be the value of the defaultLevel property (§3.27.8) of the rule object (§3.27) specified by this result object's ruleId property (§3.17.2) or ruleKey property (§3.17.3).
>
> In that case, if the run object (§3.12) containing this result does not include a rules property (§3.12.14), or if the run.rules property does not specify information for the rule associated with this result, or if the rule object associated with this result does not specify a defaultLevel property, then the value of the level property SHALL be considered to be "warning".

By setting the `default` as the value automatically, the `rule.defaultLevel` lookup mechanism is circumvented.

Can we remove the `default` from the `result.level` property? I would also suggest adjusting the `description` to better match the spec language regarding rule `rule.defaultLevel` lookups.

**Comment by michaelcfanning** [comment]
@lgolding your comments please. this all makes sense to me and I think it fixes an issue in the current spec.

**Comment by lgolding** [comment]
I agree with @lukecartey. This is a good analysis. I am now worried that there are other properties in the spec for which both
1. The schema specifies a default value, and
2. The spec text specifies a lookup mechanism to fall back to a property at run level.

So I'm going to fix this, and take the opportunity to look for other similar problems. It's not hard to find them; I just have to look for `"default":` in the schema file.

**Comment by lgolding** [comment]
The "fix" is in fact in the schema file, not in the spec at all. I will add this to the "schema changes" document and then close this issue. There is a whole bundle of work I'll have to do in the SDK to actually apply the batch of changes we're approving to the schema, and update the SDK's code to match.

**Comment by lgolding** [comment]
No, I'm an idiot. Of _course_ there are text changes in the spec, just where @lukecartey quoted them. Ok, working on this now.

**Comment by lgolding** [comment]
@michaelcfanning @lukecartey I made the changes in the [schema changes document](https://github.com/oasis-tcs/sarif-spec/blob/master/Documents/Schema%20changes%20between%20SARIF%20v1%20and%20v2.md) and closed this issue. (Look for "105" in the document.) Ultimately I'll apply each of the changes in that document to the [schema](https://github.com/Microsoft/sarif-sdk/blob/master/src/Sarif/Schemata/Sarif.schema.json) and update the SDK to match.

**Comment by lukecartey** [comment]
For reference, I've just filed a related bug against the SARIF viewer VS extension, as it doesn't appear to support rule lookups:
https://github.com/Microsoft/sarif-sdk/issues/810
I think the feature is simply missing in this case, but it might also be related to the `default` specified on the schema.

**Comment by henryju** [comment]
Hi, sorry to comment on this old issue, but I am affected by the same issue, and I can't see the fix in the SARIF schema 2.1.0. The default value is still there:
https://github.com/oasis-tcs/sarif-spec/blob/07f3a94e0046179203548dcf6e0aed7ea00bbf7d/sarif-2.1/schema/sarif-schema-2.1.0.json#L2068
Am I missing something? Should I reopen a new issue?
